packages/components/src/custom/ErrorBoundary/ErrorBoundary.tsx

```diff
@@ -1,88 +1,106 @@
+        <StyledLink href="https://discuss.layer5.io/" target="_blank" rel="noopener noreferrer">
+          Discuss Forum
+        </StyledLink>{' '}
+        {' or '}
+        <StyledLink href="https://slack.layer5.io/" target="_blank" rel="noopener noreferrer">
+          Slack
+        </StyledLink>
+      </ErrorMessage>
+      <Box>{children}</Box>
+    </div>
+  );
 }
 
-function reportError(error: Error, info: React.ErrorInfo): void {
-  // This is where you'd send the error to Sentry, etc.
+const reportError = (error: Error, info: { componentStack: string }): void => {
+  // This is where you'd send the error to Sentry, etc
   console.log('Error Caught Inside Boundary --reportError', error, 'Info', info);
+};
+
+interface ErrorBoundaryProps {
+  children: ReactNode;
 }
 
-function ErrorBoundary({ children, ...props }: ErrorBoundaryProps): JSX.Element {
+export const ErrorBoundary: React.FC<ErrorBoundaryProps> = ({ children }) => {
   return (
-    <ReactErrorBoundary FallbackComponent={Fallback} onError={reportError} {...props}>
+    <ReactErrorBoundary FallbackComponent={Fallback} onError={reportError}>
       {children}
     </ReactErrorBoundary>
   );
-}
-
-function withErrorBoundary<P extends object>(
-  Component: React.ComponentType<P>,
-  errorHandlingProps: ErrorBoundaryProps | null
-): JSX.Element {
-  const WrappedWithErrorBoundary = (props: P): JSX.Element => (
-    <ErrorBoundary {...(errorHandlingProps ? errorHandlingProps : {})}>
-      <Component {...props} />
-    </ErrorBoundary>
-  );
-
-  return WrappedWithErrorBoundary;
-}
+};
 
-interface Props {
-  children: React.ReactNode;
+interface WithErrorBoundaryProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Component: ComponentType<any>;
+  errorHandlingProps?: ErrorBoundaryProps;
 }
 
-function withSuppressedErrorBoundary<P extends object>(
-  Component: React.ComponentType<P>
-): JSX.Element {
-  const WrappedWithErrorBoundary = (props: P & Props): JSX.Element => (
-    <ErrorBoundary FallbackComponent={() => null}>
-      <Component {...props} />
+export const WithErrorBoundary: React.FC<WithErrorBoundaryProps> = ({
+  Component,
+  errorHandlingProps = { children: null }
+}: WithErrorBoundaryProps): JSX.Element => {
+  return (
+    <ErrorBoundary {...errorHandlingProps}>
+      <Component />
     </ErrorBoundary>
   );
+};
 
-  return WrappedWithErrorBoundary;
+interface WithSuppressedErrorBoundaryProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Component: ComponentType<any>;
 }
 
-export { ErrorBoundary, withErrorBoundary, withSuppressedErrorBoundary };
-*/
+export const withSuppressedErrorBoundary: React.FC<WithSuppressedErrorBoundaryProps> = ({
+  Component
+}: WithSuppressedErrorBoundaryProps): JSX.Element => {
+  return (
+    <ReactErrorBoundary FallbackComponent={Fallback} onError={reportError}>
+      <Component />
+    </ReactErrorBoundary>
+  );
+};
```

Comment:
Are we making sure that we can pass custom fallback component if we want

Reply:
suppressed should'nt have a fallback ( thats the purpose )